--- clientserver.c	2019-06-10 21:54:06.000000000 -0700
+++ clientserver.c.new	2019-06-10 21:52:16.000000000 -0700
@@ -813,46 +813,8 @@
 		}
 	}
 
-	if (gid_list.count) {
-		gid_t *gid_array = gid_list.items;
-		if (setgid(gid_array[0])) {
-			rsyserr(FLOG, errno, "setgid %ld failed", (long)gid_array[0]);
-			io_printf(f_out, "@ERROR: setgid failed\n");
-			return -1;
-		}
-#ifdef HAVE_SETGROUPS
-		/* Set the group(s) we want to be active. */
-		if (setgroups(gid_list.count, gid_array)) {
-			rsyserr(FLOG, errno, "setgroups failed");
-			io_printf(f_out, "@ERROR: setgroups failed\n");
-			return -1;
-		}
-#endif
-#if defined HAVE_INITGROUPS && !defined HAVE_GETGROUPLIST
-		/* pw is set if the user wants all the user's groups. */
-		if (pw && initgroups(pw->pw_name, pw->pw_gid) < 0) {
-			rsyserr(FLOG, errno, "initgroups failed");
-			io_printf(f_out, "@ERROR: initgroups failed\n");
-			return -1;
-		}
-#endif
-		our_gid = MY_GID();
-	}
-
-	if (set_uid) {
-		if (setuid(uid) < 0
-#ifdef HAVE_SETEUID
-		 || seteuid(uid) < 0
-#endif
-		) {
-			rsyserr(FLOG, errno, "setuid %ld failed", (long)uid);
-			io_printf(f_out, "@ERROR: setuid failed\n");
-			return -1;
-		}
-
-		our_uid = MY_UID();
-		am_root = (our_uid == 0);
-	}
+	our_uid = MY_UID();
+	am_root = (our_uid == 0);
 
 	if (lp_temp_dir(i) && *lp_temp_dir(i)) {
 		tmpdir = lp_temp_dir(i);
@@ -1064,32 +1026,10 @@
 		}
 	}
 	p = lp_daemon_gid();
-	if (*p) {
-		gid_t gid;
-		if (!group_to_gid(p, &gid, True)) {
-			rprintf(FLOG, "Invalid daemon gid: %s\n", p);
-			return -1;
-		}
-		if (setgid(gid) < 0) {
-			rsyserr(FLOG, errno, "Unable to set group to daemon gid %ld", (long)gid);
-			return -1;
-		}
-		our_gid = MY_GID();
-	}
+	our_gid = MY_GID();
 	p = lp_daemon_uid();
-	if (*p) {
-		uid_t uid;
-		if (!user_to_uid(p, &uid, True)) {
-			rprintf(FLOG, "Invalid daemon uid: %s\n", p);
-			return -1;
-		}
-		if (setuid(uid) < 0) {
-			rsyserr(FLOG, errno, "Unable to set user to daemon uid %ld", (long)uid);
-			return -1;
-		}
-		our_uid = MY_UID();
-		am_root = (our_uid == 0);
-	}
+	our_uid = MY_UID();
+	am_root = (our_uid == 0);
 
 	addr = client_addr(f_in);
 	host = lp_reverse_lookup(-1) ? client_name(f_in) : undetermined_hostname;
